design-proposals/golden-image-delivery-and-update-pipeline.md

```diff
@@ -0,0 +1,275 @@
+
+```
+
+**After Update:** And the SSP Operator would create a DataSource that pointed to the old PVC location previously referenced before the update.
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataSource
+metadata:
+  name: fedora
+spec:
+  source:
+    pvc:
+      name: fedora
+      namespace: golden-images
+```
+
+### Transitioning OCP UI to use DataSources
+
+To maintain compatiblity, the UI must be capable of detecting boot sources for DataVolumeTemplates with both inlined sources and DataSources.
+
+This means if a template references a PVC directly within the DataVolumeTemplates, then the UI should maintain the existing behavior of being able to display that the boot source is available if the PVC exists. In addition to this, if a template references a DataSource within the DataVolumeTemplates, the UI must be able to display the boot source is available if the DataSource exists and has the ready condition set to true.
+
+## Implementation Phases
+
+**Phase1** - CDI APIs
+* DataSource API/controller
+* DataImportCron API/controller
+
+**Phase2** - SSP + Common Templates
+* Common Templates transition to DataSources for DataVolumeTemplate source
+* SSP Operator creates DataImportCrons for automatic PVC import
+
+**Phase3** - HCO and UI
+* HCO Team integrates with SSP Operator’s API for setting volume import sources
+* UI Team transitions to detecting boot source using a linked DataSource rather than looking directly for a PVC of a specific name.
+
+# API Examples
+## VM + DataVolumeTemplate + DataSource as source
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+   name: my-fedora-vm-image
+spec:
+  pvc:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 2Gi
+  dataSource:
+    name: fedora
+    namespace: golden-images 
+```
+
+## DataImportCron Example
+
+This example polls for image tags for the quay.io/kubevirt/fedora image and imports the latest image sorting tags based on semver. As new tags are introduced, the DataImportCron will import them as new PVCs and automatically manage updating the corresponding DataSource.
+
+* **source** specifies where to poll from
+* **managedDataSource** specifies the name of the corresponding DataSource this cron will manage. The controller should create this DataSource if it doesn't exist, and adopt ownership of the DataSource if it does already exist.
+* **schedule** specifies in cron format when and how often to look for new imports 
+* **garbageCollectOutdated** specifies whether old PVCs should be cleaned up after a new PVC is imported.
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataImportCron
+metadata:
+  name: fedora-image-import-cron
+  namespace: golden-images
+spec:
+  source:
+    registry:
+      url: docker://quay.io/kubevirt/fedora
+  schedule: "0 */12 * * *"
+  garbageCollectOutdated: true
+  managedDataSource: fedora
+status:
+  lastImportedPVC: quay.io/kubevirt/fedora:33
```

Comment:
That status reflects the last image pulled? Or is it the name of the pvc?
If it's the PVC okay, but it probably should be a reference as well as the PVC could be in another NS right?

Reply:
Also I'd like to see more in the status like "lastPoll: timestamp" and  "lastImport: timestamp"

Reply:
@kwiesmueller great point. I've added the timestamps as you propose as well as an example of how conditions can be used to indicate an issue has occurred during import.